### ISSUE TITLE
There should be an easy way to ensure that debug_to_file() calls aren't optimized away

### ISSUE THREAD

**Issue by steven-johnson** [halide/Halide#2931]
It's handy for debugging purposes to insert calls to debug_to_file() for images that may not be part of the final pipeline (e.g. a colorized image for visualization purposes); if these aren't part of the realized output, they get optimized away, meaning you need hackery to "include" it in the output. We should smarten this somehow so that debug_to_file() calls can be forcibly kept in the pipeline without such trickery.

**Comment by zvookin** [comment]
Often the debug expression isn't even part of the IR tree when this happens. The only reason this seems like it should work is because one is thinking of the linear flow of the C++ program that makes the IR as an inclusive program. To be actionable, this likely needs to propose an IR structure and how it is made using the frontend language.

**Comment by steven-johnson** [comment]
> To be actionable, this likely needs to propose an IR structure

Agreed; this issue is being filed to actually record a small-but-annoyance-language-glitch that crops up from time to time.

**Comment by abadams** [comment]
I'm not sure how one would infer bounds for Funcs that are never used without a human-specified dependency somewhere. You could have some sugar in Generator for making the output Func have a fake dependency on f(0, 0, 0, ...) for some f, but debug_to_file isn't going to be very interesting if it only computes a single value.
